src/readbin/fsi_benchmark.ml

```diff
@@ -0,0 +1,112 @@
+open Core_kernel.Std
+open Cmdliner
+open Bap.Std
+open Bap_plugins.Std
+
+type gt_format = [ 
+  | `unstripped_bin
+  | `symbol_file ]
+
+type result = {
+  tool_name: string;
```

Comment:
this is an abstraction violation: `tool_name` is not a part of result. In general when you have some entity and some attribute that is derived from the entity, you should keep them in separate. Example, Student and his grades are different entities, c.f., a car and wheels. In the former there is no `whole-part` relationship, in the latter where is. When you're putting data into a record this is a way to express the `whole-part` relationship. 

For example, what if at some point you would like to have a map or a table of results? What if for the same tool, you will eventually have a set of different results (depending on different parameters)? 

Also, the names. Anyone, who will read this code will have no idea, what `tp` or `fn` means, and will need to dive into the implementation to figure out what is happening